src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [loadingList, setLoadingList] = useState(true);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  const filteredTodos = todoList
+    .filter(todo => todo.title.toLowerCase()
+      .includes(query.toLowerCase()))
+    .filter(todo => {
+      switch (filter) {
+        case Filter.Active:
+          return !todo.completed;
+
+        case Filter.Completed:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+
```

Comment:
this can be moved to a separate function for cleaner code

Reply:
agreed with Denis, let's create separate function for this